File: backend/models/content.go
package models

import (
	"encoding/json"
	"time"

	"gorm.io/gorm"
)

type Content struct {
	ID           uint      `json:"id" gorm:"primaryKey"`
	CategoryID   uint      `json:"category_id"`
	CategoryName string    `json:"category_name" gorm:"-"`
	Title        string    `json:"title" gorm:"not null"`
	Body         string    `json:"body" gorm:"type:text;not null"`
	Description  string    `json:"description" gorm:"default:''"`
	Poet         string    `json:"poet" gorm:"default:''"`
	Source       string    `json:"source"`
	SourceURL    string    `json:"source_url"`
	ReadTimeSecs int       `json:"read_time_secs" gorm:"default:15"`
	Tags         string    `json:"tags" gorm:"type:text"`
	Likes        int       `json:"likes" gorm:"default:0"`
	CreatedAt    time.Time `json:"created_at"`
}

// Puzzle represents an interactive puzzle (sudoku, math, logic, word)
type Puzzle struct {
	ID           uint             `json:"id" gorm:"primaryKey"`
	PuzzleType   string           `json:"puzzle_type" gorm:"not null;index"`
	CategoryID   uint             `json:"category_id" gorm:"not null;index"`
	Title        string           `json:"title" gorm:"not null"`
	Question     string           `json:"question" gorm:"type:text;not null"`
	Answer       string           `json:"-" gorm:"type:text;not null"`
	AnswerType   string           `json:"answer_type" gorm:"not null;default:'text'"`
	Options      string           `json:"options,omitempty" gorm:"type:text"`
	Hint         string           `json:"hint" gorm:"type:text;default:''"`
	Explanation  string           `json:"explanation" gorm:"type:text;default:''"`
	Difficulty   int              `json:"difficulty" gorm:"default:1"`
	Likes        int              `json:"likes" gorm:"default:0"`
	Meta         json.RawMessage  `json:"meta" gorm:"type:jsonb;default:'{}'"`
	CreatedAt    time.Time        `json:"created_at"`
}

type Category struct {
	ID             uint             `json:"id" gorm:"primaryKey"`
	Name           string           `json:"name" gorm:"uniqueIndex;not null"`
	Icon           string           `json:"icon"`
	ColorHex       string           `json:"color_hex"`
	Priority       int              `json:"priority" gorm:"default:0"`
	ContentTableID int              `json:"content_table_id" gorm:"default:0"`
	L1Category     string           `json:"l1_category" gorm:"default:''"`
	Meta           json.RawMessage  `json:"meta" gorm:"type:jsonb;default:'{}'"`
}

type Feedback struct {
	ID        uint             `json:"id" gorm:"primaryKey"`
	DeviceID  string           `json:"device_id" gorm:"index;default:''"`
	Message   string           `json:"message" gorm:"type:text;not null"`
	Meta      json.RawMessage  `json:"meta" gorm:"type:jsonb;default:'{}'"`
	CreatedAt time.Time        `json:"created_at"`
	DeletedAt gorm.DeletedAt   `json:"-" gorm:"index"`
}

type FeedResponse struct {
	Content   []Content `json:"content"`
	Page      int       `json:"page"`
	PageSize  int       `json:"page_size"`
	Total     int64     `json:"total"`
	HasMore   bool      `json:"has_more"`
}

type ContentDetail struct {
	Content
	RelatedContent []Content `json:"related_content,omitempty"`
}

File: android/app/src/main/java/com/curio/app/data/model/Content.kt
package com.curio.app.data.model

import com.google.gson.annotations.SerializedName

data class Content(
    val id: Long = 0,
    @SerializedName("category_id")
    val categoryId: Long = 0,
    @SerializedName("category_name")
    val categoryName: String = "",
    val title: String = "",
    val body: String = "",
    val description: String = "",
    val poet: String = "",
    val source: String = "",
    @SerializedName("source_url")
    val sourceUrl: String = "",
    @SerializedName("read_time_secs")
    val readTimeSecs: Int = 15,
    val tags: String = "",
    val likes: Int = 0,
    @SerializedName("created_at")
    val createdAt: String = ""
)

data class FeedResponse(
    val content: List<Content> = emptyList(),
    val page: Int = 1,
    @SerializedName("page_size")
    val pageSize: Int = 10,
    val total: Long = 0,
    @SerializedName("has_more")
    val hasMore: Boolean = false
)

data class Category(
    val id: Long = 0,
    val name: String = "",
    val icon: String = "",
    @SerializedName("color_hex")
    val colorHex: String = "",
    val priority: Int = 0,
    @SerializedName("content_count")
    val contentCount: Long = 0,
    @SerializedName("l1_category")
    val l1Category: String = ""
)

data class CategoriesResponse(
    val categories: List<Category> = emptyList(),
    val total: Int = 0
)

data class FeedbackRequest(
    val message: String,
    @SerializedName("device_id")
    val deviceId: String = ""
)

data class FeedbackResponse(
    val success: Boolean = false,
    val message: String = "",
    val id: Long = 0
)

data class L1Group(
    val name: String = "",
    val icon: String = "",
    @SerializedName("color_hex")
    val colorHex: String = "",
    val categories: List<Category> = emptyList()
)

data class L1CategoriesResponse(
    val groups: List<L1Group> = emptyList(),
    val total: Int = 0
)

data class Puzzle(
    val id: Long = 0,
    @SerializedName("puzzle_type")
    val puzzleType: String = "",
    @SerializedName("category_id")
    val categoryId: Long = 0,
    val title: String = "",
    val question: String = "",
    @SerializedName("answer_type")
    val answerType: String = "text",
    val options: String? = null,
    val hint: String = "",
    val explanation: String = "",
    val difficulty: Int = 1,
    val likes: Int = 0
)

data class PuzzleResponse(
    val puzzles: List<Puzzle> = emptyList(),
    val total: Long = 0
)

data class ValidateResponse(
    val correct: Boolean = false,
    val explanation: String = ""
)

data class ValidateRequest(
    val answer: String
)

// ── Device Info ────────────────────────────────────────────────

data class DeviceInfo(
    val id: Long = 0,
    @SerializedName("device_id")
    val deviceId: String = "",
    @SerializedName("os_version")
    val osVersion: String = "",
    @SerializedName("app_version")
    val appVersion: String = "",
    @SerializedName("device_model")
    val deviceModel: String = "",
    val manufacturer: String = "",
    @SerializedName("screen_size")
    val screenSize: String = "",
    val language: String = "",
    val timezone: String = "",
    @SerializedName("created_at")
    val createdAt: String = "",
    @SerializedName("updated_at")
    val updatedAt: String = ""
)

data class DeviceInfoRequest(
    @SerializedName("os_version")
    val osVersion: String,
    @SerializedName("app_version")
    val appVersion: String,
    @SerializedName("device_model")
    val deviceModel: String,
    val manufacturer: String,
    @SerializedName("screen_size")
    val screenSize: String,
    val language: String,
    val timezone: String
)

// ── Comments ───────────────────────────────────────────────────

data class CommentEntry(
    val id: String = "",
    val text: String = "",
    @SerializedName("device_id")
    val deviceId: String = "",
    val email: String = "",
    @SerializedName("created_at")
    val createdAt: String = ""
)

data class CommentsResponse(
    @SerializedName("content_id")
    val contentId: Long = 0,
    val comments: List<CommentEntry> = emptyList(),
    val total: Int = 0
)

data class AddCommentRequest(
    val text: String,
    val email: String = ""
)

data class AddCommentResponse(
    val success: Boolean = false,
    val comment: CommentEntry? = null
)

// ── TTS (Text-to-Speech) ───────────────────────────────────────────
data class TtsRequest(
    @SerializedName("content_id")
    val contentId: Long? = null,
    val text: String? = null,
    val voice: String = "en-US-JennyNeural"
)
